test_conformance/extensions/cl_khr_command_buffer/command_buffer_event_sync.cpp

```diff
@@ -0,0 +1,1068 @@
+            0, nullptr, nullptr, nullptr);
+        test_error(error, "clCommandNDRangeKernelKHR failed");
+
+        error = clFinalizeCommandBufferKHR(combuf);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+        return CL_SUCCESS;
+    }
+
+    //--------------------------------------------------------------------------
+    void InitInOrderEvents(std::vector<cl_event *> &event_ptrs)
+    {
+        if (out_of_order_requested)
+        {
+            in_order_events.resize(event_ptrs.size());
+            for (size_t i = 0; i < in_order_events.size(); i++)
+            {
+                event_ptrs[i] = &in_order_events[i];
+            }
+        }
+    }
+
+    //--------------------------------------------------------------------------
+    cl_int RunRegularWaitForCombuf()
+    {
+        std::vector<cl_int> output_data(num_elements);
+
+        // if out-of-order queue requested it is necessary to secure proper
+        // order of commands
+        std::vector<cl_event *> event_ptrs = { nullptr };
+        InitInOrderEvents(event_ptrs);
+
+        cl_int error =
+            clEnqueueFillBuffer(queue, in_mem, &pattern_pri, sizeof(cl_int), 0,
+                                data_size(), 0, nullptr, event_ptrs[0]);
+        test_error(error, "clEnqueueFillBuffer failed");
+
+        error = clEnqueueCommandBufferKHR(
+            0, nullptr, command_buffer, wait_count, event_ptrs[0], &test_event);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+
+        error =
+            clEnqueueReadBuffer(queue, out_mem, CL_FALSE, 0, data_size(),
+                                output_data.data(), 1, &test_event, nullptr);
+        test_error(error, "clEnqueueReadBuffer failed");
+
+        error = clFinish(queue);
+        test_error(error, "clFinish failed");
+
+        // verify the result - result buffer must contain initial pattern
+        for (size_t i = 0; i < num_elements; i++)
+        {
+            CHECK_VERIFICATION_ERROR(pattern_pri, output_data[i], i);
+        }
+
+        return CL_SUCCESS;
+    }
+
+    //--------------------------------------------------------------------------
+    cl_int RunCombufWaitForCombuf()
+    {
+        std::vector<cl_int> output_data(num_elements);
+
+        // if out-of-order queue requested it is necessary to secure proper
+        // order of all commands
+        std::vector<cl_event *> event_ptrs = { nullptr, nullptr };
+        InitInOrderEvents(event_ptrs);
+
+        cl_int error =
+            clEnqueueFillBuffer(queue, in_mem, &pattern_pri, sizeof(cl_int), 0,
+                                data_size(), 0, nullptr, event_ptrs[0]);
+        test_error(error, "clEnqueueFillBuffer failed");
+
+        error = clEnqueueCommandBufferKHR(
+            0, nullptr, command_buffer, wait_count, event_ptrs[0], &test_event);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+
+        error = clEnqueueCommandBufferKHR(0, nullptr, command_buffer, 1,
```

Comment:
This is simultaneous use of `command_buffer` on our implementation with the previous `clEnqueueCommandBufferKHR`, so the enqueue fails as `command_buffer` hasn't been created with this flag.

Reply:
Done. Small remark: actually this test "sometimes" failed with POCL as well. After applying simultaneous flag it works correctly.